src/librustc_mir/interpret/eval_context.rs

```diff
@@ -46,7 +46,7 @@ pub struct EvalContext<'a, 'mir, 'tcx: 'a + 'mir, M: Machine<'mir, 'tcx>> {
 
     /// The number of terminators to be evaluated before enabling the infinite
     /// loop detector.
-    pub(crate) steps_until_detector_enabled: usize,
+    pub(crate) steps_until_detector_enabled: isize,
```

Comment:
Yes, this could be clearer. 

My plan is to invert this counter (e.g. `steps_since_detector_enabled`), so I can use an add, then mask off the loop detector period if steps is greater than 0. The code in `inc_step_counter` currently only works with a detector period of exactly 1 due to the saturating sub. I was waiting for some clarity on the `steps` parameter to find the best way to do this.

I'm trying to make the point at which the detector comes on a comparison with `0` since it saves a cycle or two. This is probably pretty dumb :) See previous comments regarding my affinity for premature optimization.